onnxruntime/contrib_ops/cpu/transformers/beam_search.h

```diff
@@ -2,24 +2,27 @@
 // Licensed under the MIT License.
 
 #pragma once
-#include <functional>
-#include "gsl/gsl"
 #include "core/common/common.h"
-#include "core/framework/feeds_fetches_manager.h"
 #include "core/framework/op_kernel.h"
 #include "core/providers/cpu/controlflow/utils.h"
 #include "beam_search_parameters.h"
-#include "beam_search_scorer.h"
 #include "gpt_subgraph.h"
+#include "beam_search_device_helper.h"
 
 namespace onnxruntime {
+class FeedsFetchesManager;
+
 namespace contrib {
 namespace transformers {
 
-template <typename T>
-class BeamSearch : public controlflow::IControlFlowKernel {
+using namespace onnxruntime::controlflow;  // namespace of IControlFlowKernel
+
+class BeamSearch : public IControlFlowKernel {
  public:
-  BeamSearch(const OpKernelInfo& info) : IControlFlowKernel(info) { Init(info); }
+  BeamSearch(const OpKernelInfo& info) : IControlFlowKernel(info), cuda_stream_(nullptr), dumper_(nullptr) {
+    Init(info);
```

Comment:
curious why there is a separate Init() in ctor?

Reply:
Init() has some initialization code. It is possible to merge code of Init() to ctor.